src/FileIO/FileIO.cs

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+namespace DafnyLibraries {
+  using System;
+  using System.IO;
+
+  using Dafny;
+
+  public class FileIO {
+    /// <summary>
+    /// Attempts to read all bytes from the file at the given path, and outputs the following values:
+    /// <list>
+    ///   <item>
+    ///     <term>isError</term>
+    ///     <description>
+    ///       true iff an exception was thrown during path string conversion or when reading the file
+    ///     </description>
+    ///   </item>
+    ///   <item>
+    ///     <term>bytesRead</term>
+    ///     <description>
+    ///       the sequence of bytes read from the file, or an empty sequence if <c>isError</c> is true
+    ///     </description>
+    ///   </item>
+    ///   <item>
+    ///     <term>errorMsg</term>
+    ///     <description>
+    ///       the error message of the thrown exception if <c>isError</c> is true, or an empty sequence otherwise
+    ///     </description>
+    ///   </item>
+    /// </list>
+    ///
+    /// We output these values individually because Result is not defined in the runtime but instead in library code.
+    /// It is the responsibility of library code to construct an equivalent Result value.
+    /// </summary>
+    public static void INTERNAL_ReadBytesFromFile(ISequence<char> path, out bool isError, out ISequence<byte> bytesRead,
+      out ISequence<char> errorMsg) {
+      isError = true;
```

Comment:
I like that you initialize it this way. Starting out assuming an error and not indicating success until you've explicitly observed it is a nice practice.